docs/notebooks/frequency_analysis.ipynb

```diff
@@ -0,0 +1,227 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "# Frequency analysis\n",
+    "\n",
+    "Frequency analysis refers to the study of the probability of occurrence of events. It's often used in regulatory contexts to determine design values for infrastructures. For example, your city might require that road drainage systems be able to cope with a level of rainfall that is exceeded only once every 20 years on average. This 20-year return event, the infrastructure *design value*, is computed by first extracting precipitation annual maxima from a rainfall observation time series, fitting a statistical distribution to the maxima, then estimating the 95th percentile (1:20 chance of being exceeded).\n",
+    "\n",
+    "To facilitate this type of analysis on a large number of time series from model simulations or observations, `xclim` packs a few common utility functions. In the following example, we're estimating the 95th percentile of the daily precipitation maximum over the May-October period using a Generalized Extreme Value distribution.\n",
+    "\n",
+    "Note that at the moment, all frequency analysis functions are hard-coded to operate along the `time` dimension.\n",
+    "\n",
+    "Let's first create a synthetic time series of daily precipitation."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import xarray as xr\n",
+    "from xclim.indices.stats import frequency_analysis, fit, fa, parametric_quantile\n",
+    "from xclim.indices.generic import select_resample_op\n",
+    "from xclim.core.missing import missing_pct\n",
+    "from scipy.stats import gamma, bernoulli\n",
+    "\n",
+    "# Create synthetic daily precipitation time series (mm/d)\n",
+    "n = 50 * 366\n",
+    "start = np.datetime64(\"1950-01-01\")\n",
+    "time = start + np.timedelta64(1, \"D\") *  range(n)\n",
+    "#time = xr.cftime_range(start=\"1950-01-01\", periods=n)\n",
+    "\n",
+    "# Generate wet (1) /dry (0) days, then multiply by rain magnitude.\n",
+    "wet = bernoulli.rvs(.1, size=n)\n",
+    "intensity = gamma(a=4, loc=1, scale=6).rvs(n)\n",
+    "pr = xr.DataArray(wet * intensity, dims=(\"time\",), coords={\"time\": time}, attrs={\"units\": \"mm/d\"})\n",
+    "pr"
```

Comment:
would `pr.plot()` be more interesting here?  I'm pretty visual

Reply:
It's a synthetic time series, so it probably doesn't look like a realy precip time series. But yeah, we should do that once we use real precip time series.

Reply:
ok